src/components/Form.tsx

```diff
@@ -16,41 +18,54 @@ export const Form: React.FunctionComponent<IFormtronControl> = ({
   fieldComponents,
   path,
   disabled = false,
+  layout,
 }) => {
   const { variant } = useDiagnostics(path);
   const keys = Object.keys(schema.fields);
-  const guts = keys.map((name, index) => {
-    const formId = `${name}-${index}`;
-    const propSchema = schema.fields[name];
-    if (propSchema.show) {
-      const show = evaluate(propSchema.show, value, name, true);
-      if (!show) return null;
-    }
-    if (propSchema.evalOptions) {
-      propSchema.options = evaluate(propSchema.evalOptions, value, name, []);
-    }
-    const Widget = fieldComponents[propSchema.type];
-    if (Widget === undefined) {
-      throw new Error(`No appropriate widget could be found for type "${propSchema.type}"`);
-    }
-    const el = (
-      <div key={formId}>
-        <Widget
-          id={formId}
-          value={value[name]}
-          schema={propSchema}
-          path={replaceWildcards(name, path)}
-          onChange={(val: any) => {
-            const v = { ...value, [name]: val };
-            onChange(v);
-          }}
-          fieldComponents={fieldComponents}
-          disabled={disabled}
-        />
-      </div>
-    );
-    return el;
-  });
+  const grid = layout && schema.layouts && schema.layouts[layout];
+  const guts = (
+    <div
```

Comment:
If keys is empty does this div still render?

Reply:
See below

Reply:
if keys is empty.... well... I guess it does? For now lets not worry about it and if it is an issue we'll make a bug and add a short-circuit later. But really... who would pass a schema with no fields to Formtron?

Reply:
Silly end user plugin authors :). Basically, anything the user gives us (which is a LOT, all the plugin stuff, the spec files themselves, etc) cannot be trusted. We should, on principle, basically do a `DeepPartial<IWhateverInterface>` on anything that the user supplies to us to ensure we handle user driven data gracefully everywhere. 

But agree, can handle later.